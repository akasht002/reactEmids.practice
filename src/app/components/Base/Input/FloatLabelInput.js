import React from 'react';
import './styles.css'
import FloatingLabelInput from 'react-floating-label-input';
import 'react-floating-label-input/dist/react-floating-label-input.css';

class FloatLabelInput extends React.Component {
    render() {
        return(<div className="form-group">
            <FloatingLabelInput type={this.props.type} id={this.props.id} label={this.props.label} shrink={99} value={this.props.value}
                onChange={this.props.onChange} className={this.props.className}/>
                <small id={this.props.name+"Help"} className="form-text text-muted">{this.props.help}</small>
        </div>
        );
    }
}

export default FloatLabelInput;