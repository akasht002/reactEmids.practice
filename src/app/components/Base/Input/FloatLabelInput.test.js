import React from 'react';
import FloatLabelInput from './FloatLabelInput';
import renderer from 'react-test-renderer';

test('FloatLabelInput Component Jest', () => {
  const component = renderer.create(
    <FloatLabelInput type='text' id='1' label='FirstName' shrink={99} value='John Doe' className='btn'></FloatLabelInput>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

