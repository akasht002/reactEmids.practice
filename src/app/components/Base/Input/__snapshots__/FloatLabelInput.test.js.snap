// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatLabelInput Component Jest 1`] = `
<div
  className="form-group"
>
  <div
    className="react-floating-label-input active btn"
  >
    <div
      className="container"
      style={
        Object {
          "fontSize": "inherit",
          "height": "2.19em",
        }
      }
    >
      <label
        htmlFor="1"
        style={
          Object {
            "transform": "translate3d(0, -99%, 0) scale(0.99)",
          }
        }
      >
        FirstName
      </label>
      <input
        id="1"
        onBlur={[Function]}
        onChange={undefined}
        onFocus={[Function]}
        placeholder={undefined}
        type="text"
        value="John Doe"
      />
    </div>
  </div>
  <small
    className="form-text text-muted"
    id="undefinedHelp"
  />
</div>
`;
